### ISSUE TITLE
Sub-class modality

### ISSUE THREAD

**Issue by sinhaharsh** [Open-Minds-Lab/mrQA#10]
Inherit modality class to create multi-echo modalities and normal modalities, will have to change code for compliance accordingly

**Comment by raamana** [comment]
Good idea! also, this interacts with the need to modularize the methods for checking compliance (beyond simple numerical equality of values).. this needs to be attached to parameter themselves, from a set of known methods.. so when two compatible parameters are compared, we invoke the attached methods and not do an IF condition etc
